examples/util/RSocket.h

```diff
@@ -0,0 +1,36 @@
+// Copyright 2004-present Facebook. All Rights Reserved.
+
+#pragma once
+
+#include <folly/ExceptionWrapper.h>
+#include <folly/futures/Future.h>
+#include "examples/util/TcpConnectionFactory.h"
+#include "src/Payload.h"
+#include "src/ReactiveStreamsCompat.h"
+#include "src/StandardReactiveSocket.h"
+
+using namespace ::reactivesocket;
+
+/**
+ * Simplified API for client/server
+ */
+namespace rsocket {
+
+class RSocket {
+ public:
+  static std::unique_ptr<RSocket> createClientFactory(
+      std::unique_ptr<ConnectionFactory> connection);
+  RSocket(std::unique_ptr<ConnectionFactory> connection);
+  ~RSocket();
+  /*
+   * Connect asynchronously and return a Future
```

Comment:
Reason being that there are many times when fast start isn't needed/wanted, such as when building up a connection pool. In those times I want connect() to call me back with the connection when it's ready.